engine/src/main.cpp

```diff
@@ -74,6 +74,8 @@ int main(int argc, char** argv) {
   run->add_flag("--write-output,!--no-write-output", run_options.write_output,
                 "Do (not) write any output files")
       ->envname("CLOE_WRITE_OUTPUT");
+  run->add_option("-o,--output-folder",run_options.output_path, 
+                "Explicitly provide an optional path for the output files, this also enables the creation of output files");
```

Comment:
```suggestion
                "Write output to given directory, implies --write-output");
```

Reply:
Also, I'd add an environment variable for this: CLOE_OUTPUT_PATH.

Reply:
At line ~180, you do the check and apply the implication:
```cpp
if (!run_options.output_path.empty()) {
  run_options.write_output = true;
}
```

Reply:
Hmm, actually write_output is true by default, so maybe don't set --write-output at all.

It's currently disabled on purpose by smoketests, so we still want to respect that.
But maybe add a short-hand, like `-w` to `--write-output`.

Reply:
I would propose that ``--output_path`` takes priority.
If you set ``--output_path`` it would write the output regardless of the ``--write-output`` flag.

Reply:
In `main.cpp:33`, there is the following line:
```cpp
  app.option_defaults()->always_capture_default();
```
This instructs CLI to respect the default values already present for the bound types.
These are then specified by us in `main_commands.hpp`:
```cpp
struct RunOptions {
  cloe::StackOptions stack_options;
  cloe::LuaOptions lua_options;

  std::ostream* output = &std::cout;
  std::ostream* error = &std::cerr;

  // Options
  std::string uuid;
  std::string output_path;

  // Flags:
  int json_indent = 2;
  bool allow_empty = false;
  bool write_output = true;
  bool require_success = false;
  bool report_progress = true;

  bool debug_lua = false;
  int debug_lua_port = 21110;
};
```

Reply:
You have to consider that someone will do the following:
```
export CLOE_WRITE_OUTPUT=0
export CLOE_OUTPUT_PATH=./output
```

And then when running will do `cloe-engine run -w testfile.lua` to have it write output for that specific run.

Reply:
At least, that's how I would use it.